core/src/apps/mintlayer/sign_message.py

```diff
@@ -47,6 +47,7 @@ async def sign_message(
         raise DataError(f"Unknown Address type {msg.address_type}")
 
     await confirm_signverify(
+        # TODO: decode_message does not always return a valid uft-8 string e.g. for input b'\xc1\x82)'
```

Comment:
Is it true though?
I.e. from the code:
```
def decode_message(message: bytes) -> str:
    from ubinascii import hexlify

    try:
        return bytes(message).decode()
    except UnicodeError:
        return f"hex({hexlify(message).decode()})"
```
it seems that an invalid utf-8 string will raise an exception, causing the message to be encoded in hex.

P.S. my problem was that the function allows non-printable characters in the string, which will look ugly on the screen and cause the debuglink to return an invalid layout json.
Are you referring to the same problem? Or did you actually experience `decode_message` returning an invalid utf-8 string?

Reply:
in normal python it does raise an exception, but when running on the emulator it is not raising an exception and produces a string that when printed shows ")" but when decoded still shows the original bytes and raises a TypeError in the layout when trying to display it for signing. Haven't tried it on the real device maybe it is just an emulator issue.